src/main/java/com/coreos/jetcd/internal/impl/KVImpl.java

```diff
@@ -56,9 +59,14 @@
         .setPrevKv(option.getPrevKV())
         .build();
 
-    return Util.toCompletableFuture(
+    return Util.toCompletableFutureWithRetry(
         stub.put(request),
         PutResponse::new,
+        () -> {
+          stub = connectionManager.newStubWithNewToken(KVGrpc::newFutureStub);
+          return stub.put(request);
+        },
+        Util::retry,
```

Comment:
Assuming we send two put request and we handle them asynchronously, it may happen that both will fail because of token expired, then retry will be triggered thus the token will be generated twice , correct ?